x/websocket/endpoint.go

```diff
@@ -32,6 +32,10 @@ import (
 	"github.com/gorilla/websocket"
 )
 
+// defaultUpgrader is a pre-configured instance which provides reasonable default
+// values for various WebSocket upgrade parameters.
+var defaultUpgrader = websocket.Upgrader{}
```

Comment:
I don't think you are gaining anything with this global. Just create it in the Upgrade method. It will be on the stack.